### PR TITLE
#1510 Context toolbar overflow menu is truncated at edge of viewport

### PR DESCRIPTION
**Pull request by tomlyn**
Fixes: #1510
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Review by caritaou — COMMENTED** [review]
I see a warning when I leave the context menu
context toolbar warning

image

File: canvas_modules/common-canvas/src/common-canvas/cc-context-toolbar.jsx [hunk 1]
```diff
@@ -114,12 +104,41 @@ class CommonCanvasContextToolbar extends React.Component {
...
 		const toolbarItemsCount = toolbarItems.length + overflowItemCount;
 
-		// If we have some overflow menu items, we reduce the width by one pixel
-		// which forces the overflow menu and the overflow icon to be shown.
-		const reduction = overflowMenuItems.length > 0 ? 1 : 0;
+		// If we have some overflow menu items, we reduce the width by five pixels
+		// which forces the overflow menu and the overflow icon to be shown. We
+		// use 5 pixels because this is how many are needed to make the toolbar
+		// work correcty with differnet browser magnificaitons.
+		const reduction = overflowMenuItems.length > 0 ? 5 : 0;
```
[the remark this answers is not in the document]
Comment (tomlyn) [reply canvas_modules/common-canvas/src/common-canvas/cc-context-toolbar.jsx:111]: Fixed - I centered the image. The overflow icon may be a bit narrower than the others but there isn't much I can do about that. I don't think its noticible,

**Comment by tomlyn** [comment]
@caritaou I fixed both comments you made.

File: canvas_modules/common-canvas/src/toolbar/toolbar-overflow-menu.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react";
+import PropTypes from "prop-types";
+
+// import { v4 as uuid4 } from "uuid";
+// import { Button } from "carbon-components-react";
+// import { OverflowMenuVertical16 } from "@carbon/icons-react";
```
Comment (caritaou) [inline canvas_modules/common-canvas/src/toolbar/toolbar-overflow-menu.jsx:22]: Can you remove these commented out imports?

Comment (tomlyn) [reply canvas_modules/common-canvas/src/toolbar/toolbar-overflow-menu.jsx]: Fixed

**Review by caritaou — APPROVED** [review]

**Pull request merged**